extra_data/keydata.py

```diff
@@ -155,21 +165,32 @@ def source_file_paths(self):
         from pathlib import Path
         return [Path(p) for p in paths]
 
+    def _find_attributes(self, dset):
+        """Find Karabo attributes belonging to a dataset."""
+        attrs = dict(dset.attrs)
+
+        if self.is_control and self.key.endswith('.value'):
+            # For CONTROL sources, most of the attributes are saved on
+            # the parent group rather than the .value dataset.
```

Comment:
Let's also mention here that at least currently, some overlapping attributes differ, and in that case the group level attribute seems to be the one that's right.

(I have a nasty foreboding that some day we'll come across a case where the group attribute is wrong and the dataset attribute is right, and we'll have to figure out how to wrangle that. But hopefully that's just pessimism. :crossed_fingers: )

Reply:
> (I have a nasty foreboding that some day we'll come across a case where the group attribute is wrong and the dataset attribute is right, and we'll have to figure out how to wrangle that. But hopefully that's just pessimism. 🤞 )

I don't think that's a possibility.
The attributes come from the device schema. The HDF5 datasets `.value` and `.timestamp` don't exist on the device. The parent group of those datasets, is actually (on the device) the property leaf that holds the value and attributes on the device we're saving.

So the attributes on these dataset must be copied from the parent group (or have some default). I guess the wrong default could as well be on the parent group, but I don't see how this could be wrong on the parent group and correct on the dataset.

Reply:
Good point, added that clarification.

In general though, I wouldn't worry much. I'll get in touch with the DAQ team and asked them to verify our assumptions here.

We may even be able to remove the dataset attributes entirely again and shift EXtra-data to the ones on the group. Frankly I doubt users would notice.